components/applications-service/pkg/server/server.go

```diff
@@ -233,17 +236,93 @@ func convertStorageServicesToApplicationsServices(svcs []*storage.Service) []*ap
 	services := make([]*applications.Service, len(svcs))
 	for i, svc := range svcs {
 		services[i] = &applications.Service{
-			SupervisorId: svc.SupMemberID,
-			Release:      svc.FullReleaseString(),
-			Group:        svc.Group,
-			HealthCheck:  applications.HealthStatus(applications.HealthStatus_value[svc.Health]),
-			Status:       applications.ServiceStatus(applications.ServiceStatus_value[svc.Status]),
-			Application:  svc.Application,
-			Environment:  svc.Environment,
-			Fqdn:         svc.Fqdn,
-			Channel:      svc.Channel,
-			Site:         svc.Site,
+			SupervisorId:        svc.SupMemberID,
+			Release:             svc.FullReleaseString(),
+			Group:               svc.Group,
+			HealthCheck:         convertHealthStatusToProto(svc.Health),
+			Status:              applications.ServiceStatus(applications.ServiceStatus_value[svc.Status]),
+			Application:         svc.Application,
+			Environment:         svc.Environment,
+			Fqdn:                svc.Fqdn,
+			Channel:             svc.Channel,
+			Site:                svc.Site,
+			PreviousHealthCheck: convertHealthStatusToProto(svc.PreviousHealth),
+			LastHealthChange:    calculateDurationPretty(svc.HealthUpdatedAt),
 		}
 	}
 	return services
 }
+
+// TODO @afiune add comments and tests
+func calculateDurationPretty(t time.Time) (durationPretty string) {
+	year, month, day, hour, min, sec := timeDiff(t, time.Now())
+
+	// TODO @afiune fix plurals
+	if year > 0 {
+		durationPretty += fmt.Sprintf("%d years", year)
+	}
+	if month > 0 {
+		durationPretty += fmt.Sprintf("%d months", month)
+	}
+	if day > 0 {
+		durationPretty += fmt.Sprintf("%d days", day)
+	}
+	if hour > 0 {
+		durationPretty += fmt.Sprintf("%d hours", hour)
+	}
+	if min > 0 {
+		durationPretty += fmt.Sprintf("%d mins", min)
+	}
+	if sec > 0 {
+		durationPretty += fmt.Sprintf("%d seconds", sec)
+	}
+
+	return
+}
+
+func timeDiff(a, b time.Time) (year, month, day, hour, min, sec int) {
```

Comment:
does https://golang.org/pkg/time/#Time.Sub not work?

Reply:
We need a more verbose output, I can use it and then split the string, I guess.

Reply:
I am going to try to move this code into a library so that we can use it in other parts of the system. 💯